### ISSUE TITLE
Specifying that no rules be ignored

### ISSUE THREAD

**Issue by The-Penultimate-Defenestrator** [PyCQA/pycodestyle#537]
Passing `[]` for the `ignore` parameter to a `pep8.StyleGuide()` results in no rules being applied, rather than no rules being ignored.

To reproduce:
1. Create a file full of PEP8 violations
2. Initialize `pep8.StyleGuide(ignore=[])`
3. Check the file. No errors are found

The only way around this that I've found is to set `pep8.DEFAULT_IGNORE = ''`, and then not pass `ignore`.

I'm using the `1.7.0` version, which seems to be the latest stable release. Has this been addressed in the `2.0.0` alpha release? If not, it should be, because unless I'm missing something, this seems like it's not the intended behavior.

**Comment by sigmavirus24** [comment]
No, this definitely is the intended behaviour. passing `ignore=[]` is equivalent to not passing `--ignore` on the CLI. This means that the value parsed from configuration will take precedence over the default and if there is no config, it will use the default (e.g., `pep8.DEFAULT_IGNORE`).

There's also no way for us to determine if someone else is invoking a StyleGuide and they have a different thought model of how things should work than is reality. So while attempting to keep the behaviour similar, it's very hard for us to provide two different ways of doing something similar.

**Comment by The-Penultimate-Defenestrator** [comment]
I see. Is the solution I described the recommended way to enforce all
rules, then?

**Comment by sigmavirus24** [comment]
No. The way I would recommend you enforce *all* rules (including the ones that contradict each other) is:

```py
pep8.StyleGuide(select=['E', 'W'])
```

**Comment by The-Penultimate-Defenestrator** [comment]
I wasn't aware of the existence of contradictory rules, how are those
resolved?

**Comment by jayvdb** [comment]
They are resolved with the default ignore list, which is why an empty ignore list doesnt make much sense.

**Comment by IanLee1521** [comment]
Closing as not a bug.
